src/snowcli/cli/project/config.py

```diff
@@ -0,0 +1,123 @@
+import os
+import re
+from pathlib import Path
+from strictyaml import (
+    MapCombined,
+    Seq,
+    Optional,
+    UniqueSeq,
+    Bool,
+    Str,
+    Any,
+    Regex,
+    load,
+    as_document,
+)
+from collections import OrderedDict
+
+IDENTIFIER = r'(?:("[^"]*(""[^"]*)*")|([A-Za-z_][\w$]{0,254}))'
+SCHEMA_AND_NAME = f"{IDENTIFIER}[.]{IDENTIFIER}"
+GLOB_REGEX = r"^[a-zA-Z0-9_\-./*?**\p{L}\p{N}]+$"
+RELATIVE_PATH = r"^[^/][\p{L}\p{N}_\-.][^/]*$"
+
+# TODO: use the above regexes to validate paths + globs
```

Comment:
Leaving this TODO in as actually resolving paths comes later; we can verify the regexes work then